dbms/src/Interpreters/DatabaseAndTableWithAlias.h

```diff
@@ -34,6 +35,18 @@ struct DatabaseAndTableWithAlias
 
     /// Check if it satisfies another db_table name. @note opterion is not symmetric.
     bool satisfies(const DatabaseAndTableWithAlias & table, bool table_may_be_an_alias);
+
+    bool operator<(const DatabaseAndTableWithAlias& rhs) const
+    {
+        if (database == rhs.database)
+            if (table == rhs.table)
+                return alias < rhs.alias;
+            else
+                return table < rhs.table;
+        else
+            return database < rhs.database;
+
```

Comment:
`std::tie(database, table, alias) < std::tie(rhs.database, rhs.table, alias)` is simpler

Reply:
@azat 
Thank you for your comment.